swap_meet/vendor.py

```diff
@@ -1,2 +1,103 @@
-class Vendor:
-    pass
\ No newline at end of file
+from operator import invert
+from tokenize import maybe
+
+from swap_meet.item import Item
+
+class Vendor:    
+    def __init__(self,inventory=None): 
+        #super(). __init__ (category)
+        if not inventory:
+            inventory=[]
+        self.inventory=inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False  
+
+    def get_by_category(self, category):
+        
+        items = [] #creating a list where i am going to store all my objects with selected category
+        for i in self.inventory:
+            if category==i.category:
+                items.append(i)
+        return items
+       
+    def swap_items(self,friend,my_item,their_item):
+       
+        if my_item not in self.inventory or their_item not in friend.inventory:# checking if items are in inventories
+            return False
+
+        self.remove(my_item) #  if my item is in my inventory it will be removed from there and added to my friend's inventory
+        friend.remove(their_item) 
+
+        self.add(their_item) #removed item from friend's inventory is added to my inventory
+        friend.add(my_item)
+
+        return True
+
+    def swap_first_item(self,friend):
+        if self.inventory==[] or friend.inventory==[]: #checking if inventory is empty
+            return False  
+
+        my_first_item=self.inventory[0]   #first item in my inventory 
+        friends_first_item=friend.inventory[0]  #first item in friend's inventory
+
+        self.remove(my_first_item) # removing first object from my inventory 
+        friend.remove(friends_first_item)  
+
+        self.add(friends_first_item) # adding first object in friend's inventory to my inventory
+        friend.add(my_first_item) 
+        return True    
+
+    def get_best_by_category(self, category):
+        matching_category_item_list=[] # list to store all the objects with specific category
+
+        for i in self.inventory: # iterating over each object in inventory 
+            if category==i.category: # checking if the category is equal to category in inventory
+                matching_category_item_list.append(i) # add that object with matching category to the list 
```

Comment:
This section is doing the same work as `get_by_category`.  Could you use that method here?